analyzers/tests/SonarAnalyzer.Test/TestCases/AspNet/AvoidUnderPosting.cs

```diff
@@ -0,0 +1,181 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Basics
+{
+    public class ClassNotUsedInRequests
+    {
+        int ValueProperty { get; set; }                                             // Compliant
+    }
+
+    public class ModelUsedInController
+    {
+        public int ValueProperty { get; set; }                                      // Noncompliant {{Property used as input in a controller action should be nullable or annotated with the Required attribute to avoid under-posting.}}
+//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        public int? NullableValueProperty { get; set; }
+        [Required] public int RequiredValueProperty { get; set; }
+        [ValidateNever] public int NotValidatedValueProperty { get; set; }
+        [Range(0, 10)] public int ValuePropertyWithRangeValidation { get; set; }    // Noncompliant
+        [Required] public int? RequiredNullableValueProperty { get; set; }
+        public int PropertyWithPrivateSetter { get; private set; }
+        protected int ProtectedProperty { get; set; }
+        internal int InternalProperty { get; set; }
+        protected internal int ProtectedInternalProperty { get; set; }
+        private int PrivateProperty { get; set; }
+        public int ReadOnlyProperty => 42;
+        public int field = 42;
+
+        public (int, int) ImplicitTuple { get; set; }                               // Noncompliant
```

Comment:
I'm not sure if model binding supports tuples at all. You may want to add a comment about the support for Tuples in MVC binding.

Reply:
You're right, not by default. Only by creating a custom binder will it work.